packages/contracts-bedrock/src/L1/opcm/OPContractsManagerV2.sol

```diff
@@ -698,6 +711,31 @@ contract OPContractsManagerV2 is ISemver, OPContractsManagerUtilsCaller {
             revert OPContractsManagerV2_SuperchainConfigNeedsUpgrade();
         }
 
+        // Chains prior to OPCMv2 don't yet have a functional lastUsedOPCMVersion function on the
+        // SystemConfig contract. The first deployment of OPCMv2 will make this function available
+        // and subsequent deployments will then use this function to verify the system is
+        // progressing from one OPCM to the next. We only care about this for upgrades, you can
+        // perform an initial deployment from any OPCM.
+        if (!_isInitialDeployment && SemverComp.gte(_version(), "7.0.0")) {
+            ISemver lastUsedOPCM = ISemver(address(_cts.systemConfig.lastUsedOPCM()));
+            SemverComp.Semver memory lastUsedSemver = SemverComp.parse(lastUsedOPCM.version());
+            SemverComp.Semver memory thisSemver = SemverComp.parse(_version());
+
+            // We have three permitted cases:
+            // 1. Address of the last used OPCM is identical to the address of this OPCM.
+            // 2. This OPCM version is the same major version but a greater minor version.
+            // 3. This OPCM version is the next major version.
+            bool isSameOPCM = address(lastUsedOPCM) == address(thisOPCM);
+            bool isNextMajor = thisSemver.major == lastUsedSemver.major + 1;
+            bool isSameMajorHigherMinor =
+                thisSemver.major == lastUsedSemver.major && thisSemver.minor > lastUsedSemver.minor;
+
+            // Revert if none of the permitted cases are true.
+            if (!(isSameOPCM || isSameMajorHigherMinor || isNextMajor)) {
```

Comment:
Maybe its just me, but I found this hard to read.
WDYT about either:

```suggestion
            if (!isSameOPCM && !isSameMajorHigherMinor && !isNextMajor) {
```

or even just:

```suggestion
            if ((isSameOPCM || isSameMajorHigherMinor || isNextMajor) == false) {
```

Honestly the most import thing is that just that this is properly tested, and I can live with this as is.